### PR TITLE
Move integration dev docs to elastic.co

### PR DESCRIPTION
**Pull request by bmorelli25**
### Summary

We've recently made the following improvements to the [Integrations Developer Guide](https://www.elastic.co/guide/en/integrations-developer/current/index.html):

- https://github.com/elastic/observability-docs/pull/3845
- https://github.com/elastic/observability-docs/pull/3836
- https://github.com/elastic/observability-docs/pull/3792
- https://github.com/elastic/observability-docs/pull/3797
- https://github.com/elastic/observability-docs/pull/3875

With these changes, we'd like to remove the markdown files that currently live in the `/docs` dir. All of this documentation (and now more) lives in the integrations developer guide. We'd like the integrations developer guide to be the source of truth for this content.

**The goal of this project is to provide more visibility to our integration developer documentation and to add a structure/hierarchy for folks who want to learn how the process works.**

### Open questions

Does it make sense to move the source files for this content from elastic/observability-docs to elastic/integrations (this repo)? That would be a relatively easy change and it probably makes more sense for the source files to live in this repo.

### Mapping of moved content

| Old md file                                         | New docs location                                                                                 |
|-----------------------------------------------------|---------------------------------------------------------------------------------------------------|
| dashboard_guidelines.md                             | https://www.elastic.co/guide/en/integrations-developer/current/dashboard-guidelines.html          |
| definitions.md                                      | https://www.elastic.co/guide/en/integrations-developer/current/integration-definitions.html       |
| developer_tsdb_migration_guidelines.md              | https://www.elastic.co/guide/en/integrations-developer/current/developer-tsds-guidelines.html     |
| developer_workflow_bug_fix_older_package_version.md |             https://www.elastic.co/guide/en/integrations-developer/current/developer-workflow-support-old-package.html                                                                                      |
| developer_workflow_design_build_test_integration.md | https://www.elastic.co/guide/en/integrations-developer/current/build-a-new-integration.html       |
| developer_workflow_fleet_ui.md                      |             https://www.elastic.co/guide/en/integrations-developer/current/developer-workflow-fleet-UI.html                                                                                      |
| documentation_guidelines.md                         | https://www.elastic.co/guide/en/integrations-developer/current/documentation-guidelines.html      |
| fine_tune_integration.md                            |                             https://www.elastic.co/guide/en/integrations-developer/current/tips-for-building.html                                                                      |
| generic_guidelines.md                               | https://www.elastic.co/guide/en/integrations-developer/current/integrations-guidelines.html       |
| how_to_test_new_indexing_features.md                | https://www.elastic.co/guide/en/integrations-developer/current/testing-new-indexing-features.html |
| import_from_beats.md                                |      https://www.elastic.co/guide/en/integrations-developer/current/developer-workflow-import-beat.html                                                                                             |
| testing_and_validation.md                           | https://www.elastic.co/guide/en/integrations-developer/current/testing-and-validation.html        |
| tips_for_building_integrations.md                   |        https://www.elastic.co/guide/en/integrations-developer/current/developer-workflow-import-beat.html                                                                                           |

### Stakeholders / Participants / Folks that might be interested 

@nimarezainia @kilfoyle @alaudazzi @lalit-satapathy @agithomas @elastic/ecosystem

### CODE REVIEW

**Comment by kilfoyle** [comment]
Since integrations aren't specific to Observability it seems to me a very good idea to give these docs into their own home in the integrations repo.

File: docs/dashboard_guidelines.md [hunk 1]
```diff
@@ -1,132 +0,0 @@
-# Dashboard guidelines
```
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/dashboard_guidelines.md:1]: Great idea 👍

[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/dashboard_guidelines.md:1]: Done in d68b0ee19e1f5fdff21ccd2e52bac9409334e66e

[the remark this answers is not in the document]
Comment (jsoriano) [reply docs/dashboard_guidelines.md:1]: Thanks!

**Review by jsoriano — APPROVED** [review]
:+1: :+1:

**Comment by bmorelli25** [comment]
@elasticmachine test this

**Comment by bmorelli25** [comment]
Does anyone know the command to rerun the failing test?

**Comment by bmorelli25** [comment]
> Does anyone know the command to rerun the failing test?

Friendly bump on this question

**Comment by agithomas** [comment]
@bmorelli25 , please use the comment `/test` in such a situation.

**Comment by agithomas** [comment]
> Friendly bump on this question

Apologies, I missed seeing your comment.

**Comment by bmorelli25** [comment]
ugh. sorry everyone.

**Pull request merged**